examples/pytorch_distributed_simple.py

```diff
@@ -0,0 +1,184 @@
+"""
+Optuna example that optimizes multi-layer perceptrons using PyTorch distributed.
+
+In this example, we optimize the validation accuracy of hand-written digit recognition using
+PyTorch distributed data parallel and MNIST. We optimize the neural network architecture as well
+as the optimizer configuration. As it is too time consuming to use the whole MNIST dataset, we
+here use a small subset of it.
+
+You can execute this example with mpirun command as follows:
+    $ mpirun -n 2 -- python pytorch_distributed_simple.py
```

Comment:
I'm not familiar with `mpirun`, but is it necessary `--`? Since I ran this example on a GPU server, the command did not work with `--`: 

```bash
[mpiexec@a089] match_arg (../../utils/args/args.c:254): unrecognized argument
[mpiexec@a089] HYDU_parse_array (../../utils/args/args.c:269): argument matching returned error
[mpiexec@a089] parse_args (../../ui/mpich/utils.c:4770): error parsing input array
[mpiexec@a089] HYD_uii_mpx_get_parameters (../../ui/mpich/utils.c:5106): unable to parse user arguments

Usage: ./mpiexec [global opts] [exec1 local opts] : [exec2 local opts] : ...

```

Reply:
The difference may come from the implementation of MPI.
I guess you used [MPICH](https://www.mpich.org/), but I and the CI job use [Open MPI](https://www.open-mpi.org/). I'll check the behavior with MPICH too. Thank you for your feedback.

https://github.com/optuna/optuna/blob/b4450efc8239f1d180c023c9cee92cf675932bee/.github/workflows/examples.yml#L36

Reply:
I got it. Thank you for your answer!

Reply:
I reproduced the error with MPICH and updated the example for it (commit: https://github.com/optuna/optuna/pull/2303/commits/a95f239b4eebb1406730d0c7879e88fa66fd2846).

Please let me know if it does not work in your environment.

Reply:
It works now 👍